### PR TITLE
AV-84193: Change database to cluster in Capella devex docs

### PR DESCRIPTION
**Pull request by simon-dew**
Doc ticket: AV-84193

This PR changes "database" to "cluster" in the devex docs. I've followed these rules, in descending order of priority:

* If the term is "cluster" in the UI, change to cluster. (So n.b. "cluster settings", "cluster access credentials"!)
* If we're talking about the architecture of a deployment (e.g. "a node in the database"), change to cluster
* If it says "database" in the equivalent devex doc page for server, keep as database
* If it's a generic term, e.g. "a relational database", keep as database
* If referring to the generic set of data rather than the architecture of a deployment (e.g. "a document in the database"), keep as database

I have deliberately made no changes to dotted (hidden) AsciiDoc files, which need to be completely overhauled (e.g. the JavaScript UDF guides, the tutorial). I have made no changes to example files, and very sparing changes to diagram files.

Also removed one link to the Get Capella Host Name page, which is now hidden.

### CODE REVIEW

**Review by bfavini — APPROVED** [review]
This all looks fine...at some point, we may want or need to revisit to see whether we need to add the "operational" qualifier (Capella operational cluster...) But this is fine at this point.

File: modules/search/pages/create-custom-analyzer.adoc [hunk 1]
```diff
@@ -14,8 +14,8 @@ For more information, see xref:customize-index.adoc#analyzers[Search Index Featu
 
 == Prerequisites 
 
-* You have the Search Service enabled on a node in your database.
-For more information about how to change Services on your database, see xref:cloud:clusters:modify-database.adoc[].
+* You have the Search Service enabled on a node in your cluster.
+For more information about how to change Services on your cluster, see xref:cloud:clusters:modify-cluster.adoc[].
```
Comment (sarahlwelton) [inline modules/search/pages/create-custom-analyzer.adoc]:
```suggestion
For more information about how to change Services on your cluster, see xref:cloud:clusters:modify-database.adoc[].
```

**Review by sarahlwelton — CHANGES_REQUESTED** [review]
You ended up breaking one of the links :) Please fix!

File: modules/search/pages/create-custom-analyzer.adoc (hunk 1 above)
Comment (simon-dew) [reply modules/search/pages/create-custom-analyzer.adoc]: D'oh!

**Review by RichardSmedley — APPROVED** [review]
:+1:

**Review by dgnicholson — APPROVED** [review]
Took a quick look through with a focus on the usage of "Operational Clusters" and it seems consistent with how we're using in `docs-capella`. ✅

**Pull request merged**